src/states_screens/race_gui_base.cpp

```diff
@@ -1039,6 +1039,21 @@ void RaceGUIBase::drawPlayerIcon(AbstractKart *kart, int x, int y, int w)
                                                       NULL, true);
         }
     }
+
+	//lap flag for finished karts
+	if (kart->hasFinishedRace())
+	{
+        video::ITexture *icon_lapflag = irr_driver->getTexture(FileManager::GUI_ICON, "lap_flag.png");
```

Comment:
preferably, this texture should be retrieved one and stored in a  variable, not retrieved by name on every frame

Reply:
First wamted to do that as it is also done that way for the lap flag in the top right corner, but than I saw that in this function every texture is retrieved by name so I made it this way.